File: backend-queue/entity/queue.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type Queue struct {
	ID        uint           `gorm:"column:id;primaryKey;autoIncrement" json:"id"`
	QueueNo   string         `gorm:"type:varchar(50);not null;unique" json:"queue_no"`
	Name      string         `gorm:"type:varchar(100);not null" json:"name"`
	Phone     string         `gorm:"type:varchar(20);not null" json:"phone"`
	Status    string         `gorm:"type:varchar(20);default:'waiting'" json:"status"`
	QRCode    string         `gorm:"type:text" json:"qr_code"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"deleted_at"`
}

File: frontend-queue/src/types/queue.ts
export interface Queue {
  id: number;
  queue_no: string;  // ต้องตรงกับ json tag จาก Go
  name: string;
  phone: string;
  status: string;
  qr_code?: string;
  created_at?: string;
  updated_at?: string;
}